### PR TITLE
Issue 404 - refactoring simulation.py

### PR DESCRIPTION
**Pull request by sbenthall**
This PR is for refactoring of simulation.py as per #404

Currently it:
 - adds a unit test for each method in simulation.py
 - refactors drawMeanOneLognormal to use drawLognormal

I would like to go farther and refactor more of the distribution sampling code to be more concise.
But I'm making this PR mainly to test project workflow.

### CODE REVIEW

File: HARK/simulation.py [hunk 1]
```diff
@@ -33,19 +33,13 @@ def drawMeanOneLognormal(N, sigma=1.0, seed=0):
     # Set up the RNG
     RNG = np.random.RandomState(seed)
```
Comment (MridulS) [inline HARK/simulation.py]: This isn't needed in this function as the seed is passed on to `drawLognormal`.

File: HARK/tests/test_simulation.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import unittest, sys
+
+import HARK.simulation as simulation
+
+class SimulationTests(unittest.TestCase):
+    '''
+    Tests for ....
```
Comment (MridulS) [inline HARK/tests/test_simulation.py]: sampling distribution with default seed?

PS: looks like `import sys` isn't used

File: HARK/simulation.py (hunk 1 above)
Comment (sbenthall) [reply HARK/simulation.py]: good call

File: HARK/tests/test_simulation.py (hunk 2 above)
Comment (sbenthall) [reply HARK/tests/test_simulation.py]: yep

**Comment by sbenthall** [comment]
I corrected based on @MridulS 's feedback.

Based on today's meeting, my understanding now is that `simulation.py` code is deprecated and slated to be replaced with `dolark` in the near future.

So I won't do any more refactorings in this PR. I submit this PR to be reviewed and pulled.

**Pull request merged**